python/ray/data/_internal/execution/operators/union_operator.py

```diff
@@ -55,6 +55,15 @@ def num_outputs_total(self) -> Optional[int]:
             num_outputs += input_num_outputs
         return num_outputs
 
+    def num_output_rows_total(self) -> Optional[int]:
+        total_rows = 0
+        for input_op in self.input_dependencies:
+            input_num_rows = input_op.num_output_rows_total()
+            if input_num_rows is None:
+                return None
```

Comment:
For my understanding, we return None here because if we don't know the num_output_rows_total for one of the inputs we cannot reasonably estimate the total ouput for the union operator?

Reply:
Thanks for your review! From my understanding, I keep it as the same as the function above https://github.com/ray-project/ray/blob/c8885edeeba956e2842f17954fac6eb8023e9f5b/python/ray/data/_internal/execution/operators/union_operator.py#L58 and I believe that is how we handle when we don't the number of row or ref bundles. Followed this https://github.com/ray-project/ray/commit/abbc6f70a21956736df0c7c48eb63fc50577637f

Reply:
Makes sense, thanks!